client/trino-jdbc/src/main/java/io/trino/jdbc/ConnectionProperties.java

```diff
@@ -450,6 +453,11 @@ private static Predicate<Properties> isKerberosWithoutDelegation()
         return isKerberosEnabled().and(checkedPredicate(properties -> !KERBEROS_DELEGATION.getValue(properties).orElse(false)));
     }
 
+    private static Predicate<Properties> isKerberosWithDelegation()
+    {
+        return isKerberosEnabled().and(checkedPredicate(properties -> KERBEROS_DELEGATION.getValue(properties).orElse(false)));
```

Comment:
Can we use `not` here ?

Reply:
Not sure understand we can remove this method completely, but in this case instead of : 
```
        public KerberosConstrainedDelegation()
        {
            super("KerberosConstrainedDelegation", Optional.empty(), NOT_REQUIRED, isKerberosWithDelegation(), GSSCredential.class::cast);
        }
```

we need to use:

```
super("KerberosConstrainedDelegation", Optional.empty(), NOT_REQUIRED, isKerberosEnabled().and(isKerberosWithoutDelegation().negate()), GSSCredential.class::cast);
```

Reply:
Okay. I forgot JDBC is still in Java 8.